src/connection.rs

```diff
@@ -162,6 +160,43 @@ impl<T: Read + Write + ScmSocket> HttpConnection<T> {
             .ok_or(ConnectionError::ParseError(RequestError::Overflow))
     }
 
+    /// Receive data along with optional files descriptors.
+    /// It is a wrapper around the same function from vmm-sys-util.
+    ///
+    /// # Errors
+    /// `StreamError` is returned if any error occurred while reading the stream.
+    fn recv_with_fds(&mut self) -> Result<(usize, Vec<File>), ConnectionError> {
+        let buf = &mut self.buffer[self.read_cursor..];
+        // We must allocate the maximum number of receivable file descriptors
+        // if don't want to miss any of them. Allocating a too small number
+        // would lead to the incapacity of receiving the file descriptors.
+        let mut fds = [0; SCM_MAX_FD];
+        let mut iovecs = [libc::iovec {
+            iov_base: buf.as_mut_ptr() as *mut libc::c_void,
+            iov_len: buf.len(),
+        }];
```

Comment:
Nit: It would be nice if `recv_with_fds` accepted `&mut [IoSlice]` which is ABI compatible with `iovec`. If so here we could do `[IoSlice::new(buf)];` which would be easier and safer. Would it be worth creating an issue in `vmm-sys-util` for this ?

Reply:
Oh yes I agree that'd be nice, but I think this can happen as a follow up PR.

Reply:
Yes, of course.